src/dataflow-types/src/client.rs

```diff
@@ -96,6 +102,52 @@ pub struct Peek<T = mz_repr::Timestamp> {
     pub map_filter_project: mz_expr::SafeMfpPlan,
 }
 
+impl<T: PartialEq> PartialEq for Peek<T> {
```

Comment:
Could PartialEq not be automatically derived for Peek<T>?

Reply:
You are right, this works.